pkg/controller/mongodb/replica_set_controller.go

```diff
@@ -197,6 +199,9 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 
 	currentSts := appsv1.StatefulSet{}
 	if err := r.client.Get(context.TODO(), mdb.NamespacedName(), &currentSts); err != nil {
+		if apiErrors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
 		r.log.Warnf("Error getting StatefulSet: %s", err)
 		return reconcile.Result{}, err
```

Comment:
[nit] I would prefer the following:
```golang
if !apiErrors.IsNotFound(err) {
                 r.log.Warnf("Error getting StatefulSet: %s", err)
}	
return reconcile.Result{}, err			
```

Makes it easier for me to understand that the only difference is in the logging

Reply:
yep looks good to me!